### PR TITLE
Use hermit to provide a (mostly) consistent build environment

### PR DESCRIPTION
**Pull request by ivmaykov**
Hermit-ize all dependencies except clang (because the clang hermit package doesn't work on linux, I haven't tested it on other OSes)

### CODE REVIEW

File: .github/workflows/binary-static-analysis.yml [hunk 1]
```diff
@@ -33,21 +31,25 @@ jobs:
...
 
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python-version }}
+    - name: Init Hermit
+      uses: cashapp/activate-hermit@v1
 
-    - name: Display Python version
+    - name: Install Clang
```
Comment (andozw) [inline .github/workflows/binary-static-analysis.yml:37]: Can we share the shared install steps across workflows so we don't have to change a bunch of files when making changes? https://docs.github.com/en/actions/sharing-automations/avoiding-duplication

**Review by andozw — APPROVED** [review]
awesome!

File: .github/workflows/binary-static-analysis.yml (hunk 1 above)
Comment (ivmaykov) [reply .github/workflows/binary-static-analysis.yml:37]: Good idea. I'll look into it.

Comment (ivmaykov) [reply .github/workflows/binary-static-analysis.yml:37]: How about I do this in a follow-up PR? This PR is already huge.

Comment (andozw) [reply .github/workflows/binary-static-analysis.yml:37]: > How about I do this in a follow-up PR? This PR is already huge.

def better as a separate PR

**Pull request merged**